src/lib/serviceRoutines/postUpdateContext.cpp

```diff
@@ -137,13 +138,39 @@ static void updateForward(ConnectionInfo* ciP, UpdateContextRequest* upcrP, Upda
 
   ciP->outMimeType  = JSON;
 
-  //
-  // FIXME: Forwards are done using NGSIv1 only, for now
-  //        This will hopefully change soon ...
-  //        Once we implement forwards in NGSIv2, this toJsonV1() should be like this:
-  //        TIMED_RENDER(payload = upcrP->toJson());
-  //
-  TIMED_RENDER(payload = upcrP->toJsonV1(asJsonObject));
+  std::string     verb;
+  std::string     resource;
+  std::string     tenant       = ciP->tenant;
+  std::string     servicePath  = (ciP->httpHeaders.servicePathReceived == true)? ciP->httpHeaders.servicePath : "";
+  std::string     mimeType     = "application/json";
+  std::string     out;
+  int             r;
+
+  if (upcrP->providerFormat == PfJson)
+  {
+    TIMED_RENDER(payload = upcrP->toJsonV1(asJsonObject));
+
+    verb     = "POST";
+    resource = prefix + "/updateContext";
+  }
+  else
+  {
+    std::vector<std::string>  nullFilter;
+    Entity*                   eP = upcrP->entityVector[0];
+
+    eP->renderId = false;
+
+    TIMED_RENDER(payload = eP->toJson(NGSI_V2_NORMALIZED, nullFilter, false, nullFilter));
+
+    resource = prefix + "/entities/" + eP->id + "/attrs";
+    verb     = "PATCH";
```

Comment:
I'm afraid this is contrary to specification https://github.com/telefonicaid/fiware-orion/issues/3068#issuecomment-470005604:

> * `POST /v2/op/update` will use `"update"` as `actionType` (to be consistent with NGSIv1)

That's the operation has to be used to implemente NGSIv2 forwarding.

Reply:
No really contrary to the "specification" though.
You are talking about batch ops (POST /v2/op/update). This PR only deals with the PATCH operation.

As before, once we have agreed on a way to attack the forwarding, this will either stay the same or be modified.
Before we have that agreement, nothing can be done really.